libnetwork/osl/namespace_linux.go

```diff
@@ -306,11 +306,12 @@ func createNamespaceFile(path string) (err error) {
 	// before trying to create the file.
 	gpmWg.Wait()
```

Comment:
TIL: we have a GC only to delete file paths, and it serializes concurrent calls to `(*Namespace).Destroy()` through a lock.

Investigating a bit this weirdo:

- It was introduced in https://github.com/moby/libnetwork/pull/223
- @mrjana indicates it's a workaround for kernels 3.18-4.0.1
- And the kernel fix is: https://github.com/torvalds/linux/commit/8f502d5b9e3362971f58dad5d468f070340336e1
- RHEL 7 / CentOS 7 use 3.10
- RHEL 8 / CentOS 8 use 4.18

So, I'm wondering if we can safely drop it. Otherwise, we should at least document why it exists and when we'll be able to ditch it.

Reply:
> So, I'm wondering if we can safely drop it. 

I guess that depends whether RHEL backported it to their 3.10 kernels. I'll have a look at adding a comment / TODO (but maybe for this one in a follow-up as I'd have to look "what" to document exactly 😂)

Reply:
> I guess that depends whether RHEL backported it to their 3.10 kernels.

If the original comment from @mrjana is right, then 3.10 should be unaffected.

Reply:
I'd be happy if we can remove it! I just pushed an updated version; I decided to just use a separate commit for all the TODO's (to not "pollute" the other commits with them) 👍